src/components/Dropdown.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect, useCallback } from "react";
+import SVGIcon from "@/src/components/SVGIcon";
+import {
+  DropdownContainer,
+  DropdownLabel,
+  DropdownButton,
+  DropdownMenu,
+  DropdownItem,
+  ErrorMessage,
+  IconWrapper,
+} from "@/src/components/styles/Dropdown";
+
+const Dropdown = ({
+  label,
+  name,
+  options,
+  buttonText,
+  onOptionClick,
+  required,
+  errorMessage,
+  defaultSelected,
+}) => {
+  // State to manage the open/close state of the dropdown
+  const [isOpen, setIsOpen] = useState(false);
+  // State to manage the selected option, initialized with defaultSelected if provided
+  const [selectedOption, setSelectedOption] = useState(defaultSelected || "");
+
+  // Effect to close dropdown when clicking outside of it
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (isOpen && !event.target.closest(".dropdown-container")) {
+        setIsOpen(false);
+      }
+    };
+
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, [isOpen]);
+
+  // Toggle dropdown open/close state
+  const toggleDropdown = useCallback((event) => {
+    event.preventDefault();
+    setIsOpen((prev) => !prev);
+  }, []);
+
+  // Handle option selection
+  const handleOptionClick = useCallback(
+    (option) => {
+      setSelectedOption(option);
+      setIsOpen(false);
+      onOptionClick(option);
+    },
+    [onOptionClick]
+  );
+
+  // Handle key down events for the dropdown button
+  const handleKeyDown = useCallback((event) => {
+    if (["Enter", " ", "ArrowDown"].includes(event.key)) {
+      event.preventDefault();
+      setIsOpen(true);
+      setTimeout(() => {
+        document.querySelector('[role="option"]')?.focus();
+      }, 0);
+    } else if (event.key === "Escape") {
+      setIsOpen(false);
+    }
+  }, []);
```

Comment:
I am curious about that empty array. I do not see the reason for it. Could you explain this to me please?

Reply:
The empty array ensures that the function is only created once.